src/lexer.py

```diff
@@ -290,6 +339,136 @@ def word_symbolizer(source: str, in_idx: int, error_generator: ErrorGenerator) -
 
     return idx, datatypes.TokenType.Word
 
+def try_type_literal_assignment(source: str, line: int,
+                                line_start: int, start_idx: int) -> Tuple[bool, int, Optional[datatypes.Token]]:
+    """
+    Assuming we've started a poetic type literal or number assignment, attempt to tokenize
+    the remainder of the line as a type literal - a string literal, number literal, boolean, null, or mysterious.
+    Indicates whether the rest of the line was successfully tokenized so that the caller
+    can determine how to proceed. It is an error for a poetic type literal assignment to
+    include words after the type literal.
+
+    :param source:                   String of the source file
+    :param line:                     Index of the line in the source file, used to construct location
+    :param line_start:               Index of the first character in this line, used to construct location
+    :param start_index:              The index in the source file to start tokenizing from
+    :return:                         Tuple containing whether tokenizing was successful,
+                                     the next index to tokenize from, and the token produced.
+                                     The index and token are undefined if tokenizing was not successful.
```

Comment:
Following this rationale, the return type of this function should really be `Optional[Tuple[bool, int]]` which makes things well defined.

Reply:
Good suggestion, thanks!